drivers/fpga/fpga_altera.c

```diff
@@ -0,0 +1,463 @@
+		return -EBUSY;
+	}
+
+	return 0;
+}
+
+/* Validate the Reconfig status response */
+static int32_t fpga_reconfig_status_validate(struct fpga_config_status *reconfig_status_resp)
+{
+	uint32_t ret = 0;
+
+	/* Check for any error */
+	ret = reconfig_status_resp->state;
+	if (ret == MBOX_CFGSTAT_VAB_BS_PREAUTH) {
+		return MBOX_CONFIG_STATUS_STATE_CONFIG;
+	}
+
+	if (ret && ret != MBOX_CONFIG_STATUS_STATE_CONFIG)
+		return ret;
+
+	/* Make sure nStatus is not 0 */
+	ret = reconfig_status_resp->pin_status.pin_status;
+	if (!(ret & RECONFIG_PIN_STATUS_NSTATUS))
+		return MBOX_CFGSTAT_STATE_ERROR_HARDWARE;
+
+	ret = reconfig_status_resp->soft_function_status;
+	if ((ret & RECONFIG_SOFTFUNC_STATUS_CONF_DONE) &&
+	    (ret & RECONFIG_SOFTFUNC_STATUS_INIT_DONE) && !reconfig_status_resp->state)
+		return 0; /* Configuration success */
+
+	return MBOX_CONFIG_STATUS_STATE_CONFIG;
+}
+
+/* Will send the SMC command to check the status of the FPGA */
+static int32_t is_fpga_config_not_ready(void)
+{
+	uint32_t smc_cmd[2] = {0};
+	int32_t ret = 0;
+	struct private_data_t priv_data;
+
+	/* Initialize the semaphore */
+	k_sem_init(&(priv_data.smc_sem), 0, 1);
+
+	smc_cmd[SMC_REQUEST_A2_INDEX] = FPGA_CONFIG_STATUS;
+	smc_cmd[SMC_REQUEST_A3_INDEX] = 0;
+
+	/* Sending the FPGA config status mailbox command */
+	ret = smc_send(SIP_SVC_PROTO_CMD_ASYNC, SMC_FUNC_ID_MAILBOX_SEND_COMMAND, smc_cmd,
+		       &priv_data);
+	if (ret) {
+		LOG_ERR("Failed to Send the Mailbox Command !!");
+		return -ECANCELED;
+	}
+
+	k_sem_take(&(priv_data.smc_sem), K_FOREVER);
+
+	/* Verify the SMC response */
+	if (!priv_data.response.resp_data_size &&
+	    priv_data.mbox_response_len != FPGA_CONFIG_STATUS_RESPONSE_LEN) {
+		return -EINVAL;
+	}
+
+	/* Verify the FPGA config status response */
+	ret = fpga_reconfig_status_validate(
+		(struct fpga_config_status *)priv_data.mbox_response_data);
+
+	k_free(priv_data.mbox_response_data);
+
+	return ret;
+}
+
+static int32_t socfpga_bridges_reset(uint32_t enable)
+{
+	uint32_t smc_cmd[2] = {0};
+	int ret = 0;
+	struct private_data_t priv_data;
+
+	/* Initialize the semaphore */
+	k_sem_init(&(priv_data.smc_sem), 0, 1);
+
+	smc_cmd[SMC_REQUEST_A2_INDEX] = FIELD_GET(BIT(0), enable);
+
+	smc_cmd[SMC_REQUEST_A2_INDEX] |= BIT(1);
+	smc_cmd[SMC_REQUEST_A3_INDEX] = BRIDGE_MASK;
+
+	ret = smc_send(SIP_SVC_PROTO_CMD_SYNC, SMC_FUNC_ID_SET_HPS_BRIDGES, smc_cmd, &priv_data);
+	if (ret) {
+		LOG_ERR("Failed to send the smc Command !!");
+		return ret;
+	}
+
+	/* Wait for the SiP SVC callback */
+	k_sem_take(&(priv_data.smc_sem), K_FOREVER);
+
+	/* Check error code */
+	if (priv_data.response.a0) {
+		ret = -ENOMSG;
+	}
+
+	return ret;
+}
+
+static int altera_fpga_on(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	int32_t ret = 0;
+
+	/* Opening SIP SVC session */
+	ret = svc_client_open();
+	if (ret) {
+		LOG_ERR("Client open Failed!");
+		return ret;
+	}
+
+	/* Check FPGA status before bridge enable/disable */
+	ret = is_fpga_config_not_ready();
```

Comment:
I would suggest using a name that implies non-inverted logic

Reply:
done